docs/consume-packages/Package-References-in-Project-Files.md

```diff
@@ -113,10 +113,16 @@ Allowable values for these tags are as follows, with multiple values separated b
 | analyzers | .NET analyzers |
 | native | Contents of the `native` folder |
 | none | None of the above are used. |
-| all | All of the above (except `none`) |
+| all | All of the above (except `none`). Also turns off "Publish" by default. |
```

Comment:
Is this documented anywhere on the SDK side?

Personally, I've always thought this was an incorrect decision that we unfortunately can't roll back, but it is still an SDK side decision, not NuGet. So I think the NuGet docs should reference an SDK side explanation.

Reply:
I don't think it's documented anywhere. I'm not sure where the code that does this is, but I imagine most customers who are looking for something like PrivateAssets="all" will find the NuGet docs, so I think we do need to mention it here.

I don't know where I'd add that in any SDK docs, so although I'm not against having it there somewhere there and having a link to that here, I'd probably have to dig around a bit to find where an appropriate place might be—unless you already have somewhere in mind for that?

Reply:
I think publish docs should have it.

Reply:
Are you talking about https://learn.microsoft.com/dotnet/core/deploying/? I don't see anything that connects to how metadata (like PrivateAssets) would affect publishing...since that page was mostly talking about high-level things (what can you pass to the publish command), I feel like this would belong on a different page, but I didn't see anything better.

Reply:
I think any doc that may be talking about publish actually does.

The PackageReference attribute Publish is a publish first thing, not a PackageReference first thing.
It never  flows through the assets file. 

PrivateAssets means what my consumers get, it never affects what gets consumed at build time. You can take a compile time or runtime dependency on a package that you've marked with PrivateAssets.
ExcludeAssets/IncludeAssets handle whether you're allowed to take that dependency.

Publish is different from build. 

That being said, i think this discussion is related to https://github.com/dotnet/sdk/issues/39400 now.

Reply:
My understanding from that discussion is that this is still correct and unlikely to change. I couldn't find anywhere on the SDK side that documented any part of this scenario. Are you ok with merging this?